### PR TITLE
Fix regression around low row limit for CSV exports

### PR DESCRIPTION
**Pull request by mistercrunch**
Related: https://github.com/apache/incubator-superset/pull/5413 & https://github.com/apache/incubator-superset/pull/5392
@timifasubaa @villebro @john-bodley @betodealmeida

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -2354,6 +2354,15 @@ def results(self, key):
...
+        payload = utils.zlib_decompress_to_string(blob)
+        display_limit = app.config.get('DISPLAY_MAX_ROW', None)
+        if display_limit:
+            payload_json = json.loads(payload)
+            payload_json['data'] = payload_json['data'][:display_limit]
+        return json_success(
+            json.dumps(
+                payload_json, default=utils.json_iso_dttm_ser, ignore_nan=True))
+
         return json_success(utils.zlib_decompress_to_string(blob))
```
Comment (betodealmeida) [inline superset/views/core.py]: Remove this line?

Comment (mistercrunch) [reply superset/views/core.py]: oopsy

File: tests/core_tests.py [hunk 2]
```diff
@@ -634,6 +633,7 @@ def test_import_csv(self):
             os.remove(filename)
 
     def test_dataframe_timezone(self):
+        import psycopg2
```
Comment (hughhhh) [inline tests/core_tests.py]: nit: `import tz from psycopg2`

**Comment by mistercrunch** [comment]
Addressed comments, merging.

**Pull request merged**